tests/system_test.py

```diff
@@ -173,19 +173,37 @@ def get_local_host_socket(socket_address_family='IPv4'):
     return s, host
 
 
-def port_available(port, socket_address_family='IPv4'):
+def port_refuses_connection(port, socket_address_family='IPv4'):
     """Return true if connecting to host:port gives 'connection refused'."""
     s, host = get_local_host_socket(socket_address_family)
-    available = False
     try:
         s.connect((host, port))
-    except socket.error as e:
-        available = e.errno == errno.ECONNREFUSED
-    except:
-        pass
+    except OSError as e:
+        return e.errno == errno.ECONNREFUSED
+    finally:
+        s.close()
+
+    return False
+
 
-    s.close()
-    return available
+def port_permits_binding(port, socket_address_family='IPv4'):
+    """Return true if binding to the port succeeds."""
+    s, _ = get_local_host_socket(socket_address_family)
+    host = ""
+    try:
+        s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)  # so that followup binders are not blocked
+        s.bind((host, port))
+    except OSError:
+        return False
+    finally:
+        s.close()
+
+    return True
+
+
+def port_available(port, socket_address_family='IPv4'):
```

Comment:
I really like the old school convention of is_port_available() but I guess PEP-8 does not lay down any specific rules

Reply:
I guess the is/can/has convention is a good one. I'll rename the methods here to use it.

Reply:
renamed

Reply:
Please feel free to merge. Approved !